app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
+        int i = 0;
         for (String key : sEmojiConverterMap.keySet()) {
-            replaceEditable(input, key, sEmojiConverterMap.get(key));
+            if (replaceEditable(input, key, sEmojiConverterMap.get(key))){
+                i++;
+            }
+        }
+        if (i>0){
+            return true;
         }
+        else return false;
     }
 
-    private static void replaceEditable(Editable text, String in, String out) {
-        int position = text.toString().indexOf(in);
-        if (position >= 0) {
+    private static boolean replaceEditable(Editable text, String in, String out) {
```

Comment:
We need to know if something in the string has been replaced (for restarting the IME, see above...). That's why I used a bool as return value.